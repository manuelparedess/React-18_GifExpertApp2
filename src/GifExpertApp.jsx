import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifExpertApp = ({ defaultCategories = [] }) => {

    //const categories = ['Real Madrid', 'Barcelona', 'Atletico Madrid'];
    
    //const [categories, setCategories] = useState(['Real Madrid']);
    const [categories, setCategories] = useState(defaultCategories);

    //Para Agregar una categoria
    const handleAdd = (newCategory) => {

        if( categories.includes(newCategory) ) return; //Para que no se repitan las categorias

        setCategories([ newCategory, ...categories]);
    };

    return (
        <>
            <h1>Gif Expert App</h1>
            <AddCategory 
                //setCategories={setCategories} 
                onNewCategory={ handleAdd }
                //Para manejar el estado en un solo componente(el padre)
            />
            <hr />



            {
                categories.map(category => (
                    <GifGrid
                        key={category}
                        category={category}
                    />
                ))
            }
  
        </>
    );
};

export default GifExpertApp;