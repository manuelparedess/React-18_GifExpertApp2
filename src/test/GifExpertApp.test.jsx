import { render, screen } from "@testing-library/react";
import GifExpertApp from "../GifExpertApp";



describe('Pruebas en GifExpertApp', () => { 

    test('debe de mostrarse correctamente', () => {  

        const { container } = render(<GifExpertApp />);

        expect(container).toMatchSnapshot();
    });

})