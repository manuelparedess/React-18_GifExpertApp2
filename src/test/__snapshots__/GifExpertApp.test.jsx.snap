// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Gif Expert App
  </h1>
  <form
    aria-label="form"
  >
    <input
      type="text"
      value=""
    />
  </form>
  <hr />
</div>
`;
